### PR TITLE
Number of product are not ok after selecting filters

### PR DESCRIPTION
**Pull request by PierreRambaud**
Fixes https://github.com/PrestaShop/PrestaShop/issues/14734

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/ps_facetedsearch/119)

### CODE REVIEW

File: src/Adapter/AbstractAdapter.php [hunk 1]
```diff
@@ -97,10 +98,25 @@ public function resetFilter($filterName)
...
      * {@inheritdoc}
      */
-    public function resetOperationsFilter($filterName)
+    public function resetOperationsFilter($filterName, $value = null)
     {
         if ($this->operationsFilters->offsetExists($filterName)) {
-            $this->operationsFilters->offsetUnset($filterName);
+            if ($value === null) {
+                $this->operationsFilters->offsetUnset($filterName);
+            } else {
```
Comment (Progi1984) [inline src/Adapter/AbstractAdapter.php]: You can remove the else with adding a `continue` after the `offsetUnset` call

Comment (PierreRambaud) [reply src/Adapter/AbstractAdapter.php]: We can't use `continnue` outside a loop or a switch :thinking: . But I refactor the function :)

File: src/Adapter/AbstractAdapter.php [hunk 2]
```diff
@@ -97,10 +98,25 @@ public function resetFilter($filterName)
...
+                $operations = $this->operationsFilters->offsetGet($filterName);
+                foreach ($operations as $filterId => $filterValue) {
```
Comment (Progi1984) [inline src/Adapter/AbstractAdapter.php]: May be more clean

```
if (isset($operations[$value])) {
   unset($operations[$value]);
}
```

Comment (PierreRambaud) [reply src/Adapter/AbstractAdapter.php]: I want to check strictly on this index ^^

Comment (PierreRambaud) [reply src/Adapter/AbstractAdapter.php]: Oh thanks, Didn't see (came from test I think :sweat_smile: )

File: src/Adapter/AbstractAdapter.php (hunk 1 above)
Comment (Progi1984) [reply src/Adapter/AbstractAdapter.php]: #fail I see a foreach :p

File: src/Adapter/AbstractAdapter.php (hunk 2 above)
Comment (Progi1984) [reply src/Adapter/AbstractAdapter.php]: :ok_hand:

**Pull request merged**